<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Visium Image Processing</title>
  <meta name="description" content="Visium Image Processing" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Visium Image Processing" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="LieberInstitute/VisiumLIBD" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Visium Image Processing" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="step-1-split-histology-images-into-subsections.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li><a href="index.html#why-spatial-transcriptomics-or-visium-imaging">Why <em>spatial transcriptomics</em> or <em>visium imaging</em>??</a></li>
<li><a href="index.html#cite-visiumlibd">Cite <code>VisiumLIBD</code></a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="step-1-split-histology-images-into-subsections.html"><a href="step-1-split-histology-images-into-subsections.html"><i class="fa fa-check"></i><b>1</b> Step 1: Split histology images into subsections</a></li>
<li class="chapter" data-level="2" data-path="step-2-segment-nuclei-from-each-histology-subsection.html"><a href="step-2-segment-nuclei-from-each-histology-subsection.html"><i class="fa fa-check"></i><b>2</b> Step 2: Segment nuclei from each histology subsection</a>
<ul>
<li class="chapter" data-level="2.1" data-path="step-2-segment-nuclei-from-each-histology-subsection.html"><a href="step-2-segment-nuclei-from-each-histology-subsection.html#intensity-thresholding"><i class="fa fa-check"></i><b>2.1</b> Intensity thresholding</a></li>
<li class="chapter" data-level="2.2" data-path="step-2-segment-nuclei-from-each-histology-subsection.html"><a href="step-2-segment-nuclei-from-each-histology-subsection.html#kmeans-segmentation"><i class="fa fa-check"></i><b>2.2</b> Kmeans segmentation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="step-3-run-spaceranger.html"><a href="step-3-run-spaceranger.html"><i class="fa fa-check"></i><b>3</b> Step 3: Run spaceranger</a></li>
<li class="chapter" data-level="4" data-path="step-4-spotspotcheck-for-quality-check-on-nuclei-segmentations-and-counts.html"><a href="step-4-spotspotcheck-for-quality-check-on-nuclei-segmentations-and-counts.html"><i class="fa fa-check"></i><b>4</b> Step 4: spotspotcheck for quality check on nuclei segmentations and counts</a></li>
<li class="chapter" data-level="5" data-path="step-5-atlasaligner.html"><a href="step-5-atlasaligner.html"><i class="fa fa-check"></i><b>5</b> Step 5: AtlasAligner</a></li>
<li class="chapter" data-level="" data-path="r-session-information.html"><a href="r-session-information.html"><i class="fa fa-check"></i>R session information</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Visium Image Processing</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Visium Image Processing</h1>
<p class="author"><em>Madhavi Tippani</em></p>
<p class="author"><em>true</em></p>
</div>
<div id="overview" class="section level1 unnumbered">
<h1>Overview</h1>
<p><code>*Visium imaging*</code> is a part of whole <code>*spatial transcriptomics*</code> workflow, a barcoding-based transcriptome-wide technology released by <a href="https://www.10xgenomics.com">10X genomics</a>.</p>
<p><img src="images/visium_imaging.png" title="Visium Imaging" align="middle" width="500px"/>
(<a href="https://www.10xgenomics.com/products/spatial-gene-expression" class="uri">https://www.10xgenomics.com/products/spatial-gene-expression</a>)
<img src="images/visium_spot.png" title="Gene Expression Slide" align="middle" width="500px"/></p>
<div id="why-spatial-transcriptomics-or-visium-imaging" class="section level2 unnumbered">
<h2>Why <em>spatial transcriptomics</em> or <em>visium imaging</em>??</h2>
<p>Methods like RNAseq can profile single cells transcriptome-wide and enable researchers to identify cell type compositions but can necessarily destroy the spatial information. Multiplexing methods on other hand can provide spatial information but has significant limitations on the number of genes that can be processed and has issues with microscopy and related computational challenges. <code>*spatial transcriptomics*</code> provides solutions to these limitations allowing us to quantify gene expression with high spatial resolution. The key to this system is the <code>*visium imaging*</code> and <code>*visium gene expression slide*</code>, where the tissue sections from subjects goes onto the capture areas(green squares) on the slide. The whole slide is imaged at once and produces a huge output image file with all capture areas. The whole slide image has to be split into individual capture area images(necessarily JPEG or tiff) and processed accordingly for the downstream analysis.</p>
<p>This website describes the steps required to split, visualize and process the visium images from spatial transcriptomics projects generated by the <a href="https://www.10xgenomics.com/spatial-transcriptomics">10x Genomics Visium</a> commercial platform.</p>
<p><img src="images/VisiumLIBD.png" title="VisiumLIBD" align="middle" width="500px"/></p>
<p>The above figure describes the VisiumLIBD pipeline, where sections (A) and (B) correspond to visium imaging and the rest describes the workflow in a step-by-step procedure. (C) First, the whole slide image is split into the respective capture areas. (D) Then the nuclei from individual tiffs are segmented using <a href="https://www.mathworks.com/help/images/color-based-segmentation-using-k-means-clustering.html">Kmeans</a> color based segmentation (E) The output tiff files from (D) serve as input into the <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/count">Spaceranger</a> module for subsequent spatial transcriptomics processing and analysis. (F) Next we extract nuclei count per visium spot in the image. (G) Finally, the pipeline provides a GUI that allows the user to perform visual inspection of the segmentations by allowing the user to toggle between the visium and binary images, and also by providing zoom in/out options to clearly see the nuclei in the visium spot.</p>
</div>
<div id="cite-visiumlibd" class="section level2 unnumbered">
<h2>Cite <code>VisiumLIBD</code></h2>
<p>We hope that <a href="http://research.libd.org/VisiumLIBD/">VisiumLIBD</a> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<pre><code>@article {,
    author = {Tippani, Madhavi and other names to include},
    title = {VisiumLIBD: a semiautomated Matlab pipeline for visualizing and processing high resolution visium images whose output is used for downstream spatial transcriptomics analyis},
    elocation-id = {},
    year = {2021},
    doi = {},
    publisher = {Oxford Bioinformatics},
    abstract = {},
    URL = {https://www.overleaf.com/read/fmvrpsyxgsbz},
    eprint = {https://www.overleaf.com/read/fmvrpsyxgsbz},
    journal = {bioRxiv}
}</code></pre>
<p>This is a project by the <strong>Imaging Development Team</strong> from <strong>Keri Martinowich’s Lab</strong> at the <a href="https://www.libd.org/">Lieber Institute for Brain Development</a>.</p>
<table>
<tbody>
<tr class="odd">
<td><a href="https://www.libd.org/"><img src="http://lcolladotor.github.io/img/LIBD_logo.jpg" align="middle" width="250px"></a></td>
<td></td>
</tr>
</tbody>
</table>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="step-1-split-histology-images-into-subsections.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
