<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Step 4: GUI to count nuclei in a Visium spot | VistoSeg: Visium Histology Image Segmentation and Processing Pipeline</title>
  <meta name="description" content="4 Step 4: GUI to count nuclei in a Visium spot | VistoSeg: Visium Histology Image Segmentation and Processing Pipeline" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Step 4: GUI to count nuclei in a Visium spot | VistoSeg: Visium Histology Image Segmentation and Processing Pipeline" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="LieberInstitute/VistoSeg" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Step 4: GUI to count nuclei in a Visium spot | VistoSeg: Visium Histology Image Segmentation and Processing Pipeline" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="step-3-spaceranger.html"/>
<link rel="next" href="step-5-test-data-and-future-directions.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li><a href="index.html#why-spatial-transcriptomics-or-visium-imaging">Why <em>spatial transcriptomics</em> or <em>Visium imaging</em>?</a></li>
<li><a href="index.html#cite-vistoseg">Cite <code>VistoSeg</code></a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#image-acquisition"><i class="fa fa-check"></i>Image Acquisition</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#software-requirements"><i class="fa fa-check"></i>Software Requirements</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#data-availability"><i class="fa fa-check"></i>Data Availability</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html"><a href="step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html"><i class="fa fa-check"></i><b>1</b> Step 1: Split Visium histology whole slide image into individual capture area images</a></li>
<li class="chapter" data-level="2" data-path="step-2-nuclei-segmentation-of-individual-capture-areas-images.html"><a href="step-2-nuclei-segmentation-of-individual-capture-areas-images.html"><i class="fa fa-check"></i><b>2</b> Step 2: Nuclei segmentation of individual capture areas images</a>
<ul>
<li class="chapter" data-level="2.1" data-path="step-2-nuclei-segmentation-of-individual-capture-areas-images.html"><a href="step-2-nuclei-segmentation-of-individual-capture-areas-images.html#vns---visium-nuclei-segmentation"><i class="fa fa-check"></i><b>2.1</b> VNS - Visium Nuclei Segmentation</a></li>
<li class="chapter" data-level="2.2" data-path="step-2-nuclei-segmentation-of-individual-capture-areas-images.html"><a href="step-2-nuclei-segmentation-of-individual-capture-areas-images.html#refinevns"><i class="fa fa-check"></i><b>2.2</b> refineVNS</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="step-3-spaceranger.html"><a href="step-3-spaceranger.html"><i class="fa fa-check"></i><b>3</b> Step 3: spaceranger</a>
<ul>
<li class="chapter" data-level="" data-path="step-3-spaceranger.html"><a href="step-3-spaceranger.html#what-is-spaceranger"><i class="fa fa-check"></i>What is spaceranger?</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="step-4-gui-to-count-nuclei-in-a-visium-spot.html"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html"><i class="fa fa-check"></i><b>4</b> Step 4: GUI to count nuclei in a Visium spot</a>
<ul>
<li class="chapter" data-level="4.1" data-path="step-4-gui-to-count-nuclei-in-a-visium-spot.html"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#spotspotcheck"><i class="fa fa-check"></i><b>4.1</b> spotspotcheck</a></li>
<li class="chapter" data-level="4.2" data-path="step-4-gui-to-count-nuclei-in-a-visium-spot.html"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#countnuclei"><i class="fa fa-check"></i><b>4.2</b> countNuclei</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="step-5-test-data-and-future-directions.html"><a href="step-5-test-data-and-future-directions.html"><i class="fa fa-check"></i><b>5</b> Step 5: Test data and future directions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="step-5-test-data-and-future-directions.html"><a href="step-5-test-data-and-future-directions.html#public-datasets"><i class="fa fa-check"></i><b>5.1</b> Public datasets</a></li>
<li class="chapter" data-level="5.2" data-path="step-5-test-data-and-future-directions.html"><a href="step-5-test-data-and-future-directions.html#future-directions"><i class="fa fa-check"></i><b>5.2</b> Future directions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-session-information.html"><a href="r-session-information.html"><i class="fa fa-check"></i>R session information</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">VistoSeg: Visium Histology Image Segmentation and Processing Pipeline</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="step-4-gui-to-count-nuclei-in-a-visium-spot" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Step 4: GUI to count nuclei in a Visium spot</h1>
<div id="spotspotcheck" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> spotspotcheck</h2>
<p>The GUI is used for
1. Obtaining nuclei count per Visium spot and saving it to a csv.<br />
2. Performing quality check on the nuclei segmentations (from <a href="http://research.libd.org/VistoSeg/step-2-nuclei-segmentation-of-individual-capture-areas-images.html">Step2</a>) by allowing the user to zoom in/out and toggle between the nuclei segmentation and Visium histology images. The GUI overlays the spot grid on to the Visium histology and segmentation images for the user to clearly see the nuclei inside a spot.<br />
3. Allowing the user to search for a spot with its associated barcode ID.</p>
<p>To open the GUI, the <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/code/spotspotcheck.m">spotspotcheck</a> function should be executed in the MATLAB command prompt like below.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb8-1"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="va">spotspotcheck</span></span>
<span id="cb8-2"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="va">loading</span><span class="op">...</span>..</span></code></pre></div>
<ol style="list-style-type: decimal">
<li><p><strong>Start Window</strong> : A start window opens up like below. If you donot have the <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spotspotcheck/tissue_spot_counts.csv">tissue_spots_counts.csv</a> already, generated from the command line function <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/code/countNuclei.m">countNuclei</a>, the option to <code>Get Cell Counts</code> on the window should be checked.</p>
<center>
<p><img src="images/start.png" /></p>
</center>
<p>Then you hit the <code>Start</code> button, which opens up the following window.</p></li>
<li><p><strong>Select Histology Image</strong> : This window asks you to select the capture area tif that you want the nuclei counts for, or to perform quality check on nuclei segmentations.</p>
<center>
<p><img src="images/Select%20Histology%20Image.png" /></p>
</center>
<p>Select the capture area tif and hit the <code>Open</code> button on window, which opens the following window.</p></li>
<li><p><strong>Select Segmented Mat File</strong> : This wondow asks for the final segmented <code>.mat</code> file of the nuclei segmentations.</p>
<center>
<p><img src="images/Select%20segmentation.png" /></p>
</center>
<p>Select the <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/refineVNS/Liebert_Institute_OTS-20-7690_rush_anterior_A1_nuclei.mat">.mat</a> file and hit the <code>Open</code> button.</p></li>
<li><p><strong>Select Scale Factors JSON File</strong> : Select the <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spaceranger/scalefactors_json.json">scalefactors_json.json</a> file with the Visium spot metrics obtained from the <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger"><code>spaceranger</code></a> software.</p>
<center>
<p><img src="images/select%20json.png" /></p>
</center></li>
<li><p><strong>Select Tissue Positions/Spot Counts File</strong> : If the <code>Get Cell Counts</code> option is checked in the Start window then select <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spaceranger/tissue_positions_list.csv">tissue_positions_list.csv</a> obtained from <strong>Spaceranger</strong>, otherwise select <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spotspotcheck/tissue_spot_counts.csv">tissue_spot_counts.csv</a> obtained from <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/code/countNuclei.m">countNuclei</a> function.</p>
<center>
<p><img src="images/select%20spot%20counts.png" /></p>
</center></li>
<li><p><strong>spotspotcheck</strong> GUI opens up, shown below</p>
<center>
<p><img src="images/spotspotcheck.png" /></p>
</center></li>
</ol>
By clicking anywhere on the image inside the GUI, the user can toggle between the histology and segmented image, shown below.
<center>
<img src="images/spotspotcheck%20BW.png" />
</center>
The zoom in (symbol of magnifying glass with <code>+</code>) option is provided in the menubar at the top of the GUI (Note the position of this option on the GUI may change depending on the version of MATLAB being used, but it is always represented by the <code>+</code> symbol). Selecting/clicking this option opens a cursor with <code>+</code> symbol, which lets the user select a rectangular region of interest to zoom in to, like the image below. The <code>Zoom Out</code> button at the bottom of GUI is used for the images to return to their original size on GUI.
<center>
<img src="images/zoom%20in%201.png" />
</center>
<p>To toggle between histology/segmented images when zoomed in, the cursor should return to its normal function by un-clicking the zoom in (symbol of magnifying glass with <code>+</code>) option.</p>
<p><img src="images/zoom%20in%202.png" title="Color cluster 1" width="350"/> <img src="images/zoom%20in%202%20BW.png" title="Indexed object 1" width="350"/></p>
The dialog box at the bottom of the GUI with the label <code>Barcode Lookup</code> provides user to input the barcode ID of the specific Visium spot the user want to look at. For example, if we want to look at the 1st Visium spot on this image, paste the barcode ID of the spot (from <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spotspotcheck/tissue_spot_counts.csv">tissue_spot_counts.csv</a>) into the white space and hit the <code>Lookup</code> button. It opens up the image where ever this spot is located.
<center>
<img src="images/Barcode%20ID.png" />
</center>
</div>
<div id="countnuclei" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> countNuclei</h2>
<p>This function can be run on MATLAB command prompt to obtain the <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spotspotcheck/tissue_spot_counts.csv">tissue_spot_counts.csv</a> file without going through the <code>spotspotcheck</code> GUI.</p>
<p>Inputs to the function are<br />
1. Filename with full path for the mat file of the final nuclei segmentation.<br />
2. Filename with full path for the scalefactors_json json file.<br />
3. Filename with full path for the tissue_positions_list csv file.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb9-1"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="va">mask</span> <span class="op">=</span> <span class="ss">&#39;/path_to_capture_area_segmentation/Lieber_Institute_OTS-20-7690_rush_anterior_A1_nuclei.mat&#39;</span><span class="op">;</span></span>
<span id="cb9-2"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="va">jsonname</span> <span class="op">=</span> <span class="ss">&#39;/path_to_scalefactors_json/scalefactors_json.json&#39;</span><span class="op">;</span></span>
<span id="cb9-3"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="va">posname</span> <span class="op">=</span> <span class="ss">&#39;/path_to_tissue_positions_list/tissue_positions_list.csv&#39;</span><span class="op">;</span></span>
<span id="cb9-4"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="va">countNuclei</span>(<span class="va">mask</span><span class="op">,</span><span class="va">jsonname</span><span class="op">,</span><span class="va">posname</span>)  </span>
<span id="cb9-6"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="va">loading</span> <span class="va">data</span></span>
<span id="cb9-7"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="va">Elapsed</span> <span class="va">time</span> <span class="va">is</span> <span class="fl">4.216505</span> <span class="va">seconds</span>.</span>
<span id="cb9-8"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fl">84385</span> <span class="va">cells</span> <span class="va">detected</span></span>
<span id="cb9-9"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fl">4992</span> <span class="va">spots</span> <span class="va">detected</span></span>
<span id="cb9-10"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="va">counting</span> <span class="va">nuclei</span> <span class="va">per</span> <span class="va">spot</span></span>
<span id="cb9-11"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fl">100</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">77.1357</span><span class="va">s</span></span>
<span id="cb9-12"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fl">200</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">140.8369</span><span class="va">s</span></span>
<span id="cb9-13"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fl">300</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">204.5829</span><span class="va">s</span></span>
<span id="cb9-14"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="fl">400</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">267.4467</span><span class="va">s</span></span>
<span id="cb9-15"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="fl">500</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">335.6774</span><span class="va">s</span></span>
<span id="cb9-16"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="fl">600</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">403.0874</span><span class="va">s</span></span>
<span id="cb9-17"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="fl">700</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">467.4493</span><span class="va">s</span></span>
<span id="cb9-18"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fl">800</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">534.8536</span><span class="va">s</span></span>
<span id="cb9-19"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="fl">900</span> <span class="va">spots</span> <span class="va">finished</span> <span class="va">in</span> <span class="va">time</span> <span class="fl">602.1899</span><span class="va">s</span></span>
<span id="cb9-20"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-20" aria-hidden="true" tabindex="-1"></a>.</span>
<span id="cb9-21"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-21" aria-hidden="true" tabindex="-1"></a>.</span>
<span id="cb9-22"><a href="step-4-gui-to-count-nuclei-in-a-visium-spot.html#cb9-22" aria-hidden="true" tabindex="-1"></a>.</span></code></pre></div>
<p>The <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/code/countNuclei.m">countNuclei</a> function generates a nuclei count file listed below. The file is saved in the same path as the <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spaceranger/tissue_positions_list.csv">tissue_positions_list.csv</a> that we give to the function.<br />
1. <a href="https://github.com/LieberInstitute/VistoSeg/blob/main/pipeline_outputs/spotspotcheck/tissue_spot_counts.csv">tissue_spot_counts.csv</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="step-3-spaceranger.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="step-5-test-data-and-future-directions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
